### PR TITLE
feat: include "extraEnvVars" as part of the migration job

### PR DESCRIPTION
**Pull request by Oscmage**
## Description

From reading the code I see no way of providing the same `extraEnvVars` as done to the deployment for the migration job. The only way I can see is to directly add `datastore.uri` as part of the values file which is not something we want since it is a secret and something we want to commit to a git repository. It is from what I've found (please correct me if so) not possible to to use something like ExternalSecret to load the value dynamically rather than hardcoding it in the values file.

I'd like to be able to do something like

We are currently running this version of the chart in production and it works just fine.

```
openfga:
  extraEnvVars:
    - name: OPENFGA_DATASTORE_URI
      valueFrom:
        secretKeyRef:
          name: openfga-external-secrets
          key: OPENFGA_DATASTORE_URI
```
but it is currently not picked up by the migration job.

## References

How it is done in the deployment file: 

https://github.com/openfga/helm-charts/blob/40ea4e1936efb04da540d1d2afe7e6375eedda3f/charts/openfga/templates/deployment.yaml#L326

## Review Checklist
- [X] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [X] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [X] The correct base branch is being used, if not `main`
- [X] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

**Comment by Oscmage** [comment]
Found another PR for this. https://github.com/openfga/helm-charts/pull/134

File: charts/openfga/templates/job.yaml [hunk 1]
```diff
@@ -51,6 +51,11 @@ spec:
                   name: "{{ .Values.datastore.uriSecret }}"
                   key: "uri"
             {{- end }}
+
+            {{- with .Values.extraEnvVars }}
+              {{- toYaml . | nindent 12 }}
+            {{- end }}
```
Comment (Oscmage) [inline charts/openfga/templates/job.yaml:57]: Same as we do for deployment:

https://github.com/openfga/helm-charts/blob/40ea4e1936efb04da540d1d2afe7e6375eedda3f/charts/openfga/templates/deployment.yaml#L326

**Review by ewanharris — APPROVED** [review]
This makes sense to me given #134, but I'd like to tag @jon-whit to make sure there's nothing I'm missing

**Comment by Oscmage** [comment]
@ewanharris Make sense, thanks!

**Comment by Oscmage** [comment]
Resolved merge conflict due to other changes.

**Review by rhamzeh — DISMISSED** [review]

**Comment by jon-whit** [comment]
@Oscmage please don't forget to bump the `chartVersion`.

**Review by jon-whit — DISMISSED** [review]

**Comment by Oscmage** [comment]
> @Oscmage please don't forget to bump the `chartVersion`.

Sorry might have gotten lost in the merge.

@jon-whit Do you mean `charts/openfga/Chart.yaml` `version`? Did that change now.

**Review by jon-whit — APPROVED** [review]

**Pull request merged**